### ISSUE TITLE
reconnection error with v2.0.0-213

### ISSUE THREAD

**Issue by ThomasHalwax** [nats-io/nats.js#376]
Since the release of v2 is expected soon I've been playing around with a very simple publish/subscribe scenario. I'm using version 2.0.0-213 installed by ```npm install nats@beta```.

While being connected with an active subscription I shut down the server and the client lib correctly returns events via 
```javascript
const handleStatusChanged = async (connection) => {
  for await (const event of connection.status()) {
    console.log(`reveived event of type ${event.type}`)
  }
}

handleStatusChanged(connection)
```

```bash
reveived event of type disconnect
reveived event of type reconnecting
```

After a short period of time nats.js throws an exception:
```bash
(node:10288) UnhandledPromiseRejectionWarning: NatsError: TIMEOUT
    at Function.errorForCode (/Users/thomas/Development/nats/client/node_modules/nats/lib/nats-base-client/error.js:85:16)
    at Timeout._onTimeout (/Users/thomas/Development/nats/client/node_modules/nats/lib/nats-base-client/util.js:136:38)
    at listOnTimeout (internal/timers.js:531:17)
    at processTimers (internal/timers.js:475:7)
(node:10288) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 5)
(node:10288) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Obviously the connection does not get reestablished.

Thanks, Thomas

**Comment by aricart** [comment]
is your `maxReconnectAttempts` set to -1?

**Comment by ThomasHalwax** [comment]
> is your `maxReconnectAttempts` set to -1?

Not explicitly. Here's the complete code:
```javascript
const { connect, JSONCodec } = require("nats")
const payloadCodec = JSONCodec()

const handle = async (subscription) => {
  for await (const message of subscription) {
    console.log(`received ${subscription.getProcessed()}`)
    console.dir(payloadCodec.decode(message.data))
  }
}

const handleStatusChanged = async (connection) => {
  for await (const event of connection.status()) {
    console.log(`reveived event of type ${event.type}`)
  }
}

const main = async () => {
  try {
    const connection = await connect({ 
      servers: 'nats.mindlab.at', 
      name: 'client-subscriber', 
      debug: true, 
      waitOnFirstConnect: true, 
      tls: {}, 
      verbose: true 
    })
    console.log(`connected to ${connection.getServer()}`)
    console.dir(connection.info)

    connection.closed()
    .then(error => {
      console.log('connection has been closed')
      if (error) {
        console.error(error)
      }
    })
    .catch((error) => {
      console.dir(error)
    })
  
    handleStatusChanged(connection)

    const subject = 'emea.>'
    const subscription = connection.subscribe(subject)
    console.log(`subscribed to ${subject} using subscription id ${subscription.getID()}`)
    handle(subscription)
  } catch (error) {
    console.error(error)
  }
}

main()
```

**Comment by wiltonfelix** [comment]
Hello @aricart, 

I have that problem too.

My code:
`const STAN = require("node-nats-streaming");

let sc;

/**
 * Create a new connection with Nats
 * @param callback Function callback for 'connect' event
 */
export const connect = (callback = () => {}) => {
  if (sc) {
    throw new Error("nats already connected");
  }

  const log = InitLogger();
  const hash = new Date().getTime();

  sc = STAN.connect("nats-streaming", `app-${hash}`, {
    url: process.env.NATS_URL,
    user: process.env.NATS_USER,
    pass: process.env.NATS_PASS,
    json: true,
    maxPingOut: 30,
    pingInterval:5000,
    reconnect: true,
    waitOnFirstConnect: true,
    debug: true, 
    maxReconnectAttempts: -1,
    reconnectTimeWait: 250
  });

  sc.on('error', (err) => {
    console.log(err)
  })

  sc.on("connect", () => {
    console.log("Core is connected to nats-streaming");
  });
 
  sc.on('reconnecting', () => {
    console.log("Core is reconnecting to nats-streaming");
  })
  
  sc.on('reconnect', () => {
    console.log("Core is reconnected to nats-streaming");
  })
};`

**Comment by aricart** [comment]
@ThomasHalwax thanks for the report and code snippet.

**Comment by aricart** [comment]
@ThomasHalwax `npm install nats@beta` to get a fix.

**Comment by ThomasHalwax** [comment]
> @ThomasHalwax `npm install nats@beta` to get a fix.

Thanks @aricart, I can confirm that the issue is fixed!

**Comment by semireg** [comment]
I was also running into this. I am unable to install nats@beta. Where can I get this fix? I'm currently running `2.0.0-220`.

**Comment by semireg** [comment]
Note above: I am running the rc as 2.0.0-220.
